### PR TITLE
docs: improve jsonb function examples

### LINKED ISSUE #20258: docs: improvements for jsonb_agg and other jsonb functions
### Documentation request

- The jsonb_agg function is not listed as a function under the jsonb type docs
- The examples for some of the functions could be a little more practical, like eg using subqueries to define their inputs

Example:

```sql
SELECT jsonb_agg(my_input) AS my_agg FROM (SELECT * FROM my_view WHERE id > 1) AS my_input;
```
```
                                                  my_agg                                                   
-----------------------------------------------------------------------------------------------------------
 [{"content":"hi","id":2},{"content":"ho","id":3},{"content":"bloop","id":5}]
(1 row)
```

Another example:

```sql
SELECT jsonb_build_array(my_input) FROM (SELECT * FROM my_view WHERE id > 1) AS my_input;
```

```
       jsonb_build_array        
--------------------------------
 [{"content":"hi","id":2}]
 [{"content":"ho","id":3}]
 [{"content":"bloop","id":5}]
(3 rows)
```

### Affected Pages

- https://materialize.com/docs/sql/types/jsonb/#functions
- https://materialize.com/docs/sql/functions/jsonb_agg/

### Related work

_No response_

### PR DESCRIPTION
**Pull request by chuck-alt-delete**
closes #20258

Improve examples to various jsonb function docs and include jsonb aggregation functions under the jsonb type doc.

### CODE REVIEW

**Comment by chuck-alt-delete** [comment]
I'm also wondering whether to bring the descriptions and type signatures in `doc/user/data/sql_funcs.yml` more in line with the postgres docs for [json functions](https://www.postgresql.org/docs/current/functions-json.html), but that might be out of scope for this PR.

**Comment by chuck-alt-delete** [comment]
As an aside, the `url` attribute from `sql_funcs.yml` doesn't work reliably on previews. Including a link directly in the description seems to work better. The only function that does this is `list_agg`:
```yml
- type: List
  description: List functions take [`list`](../types/list) arguments, and are [polymorphic](../types/list/#polymorphism).
  functions:
    - signature: 'list_agg(x: any) -> L'
      description: Aggregate values (including nulls) as a list. [(docs)](/sql/functions/list_agg)
```
I've gone this route for the two json aggregate functions I added to the JSON section, but we may want to look into fixing the shortcode that translates the `url` attribute into a link to work in previews.

File: doc/user/content/sql/functions/jsonb_object_agg.md [hunk 1]
```diff
@@ -48,20 +48,35 @@ Instead, we recommend that you materialize all components required for the
...
 ```
 ```nofmt
- jsonb_object_agg
+      my_agg
 ------------------
- {"key1": 1, "key2": null}
+ {"k1": 1, "k2": -1}
 ```
+Notice there are multiple values that could potentially correspond to the key `k2`.
+In this case, we use a `FILTER` to look at values that are not `NULL` and order by a timestamp column to get the most recent such value.
```
Comment (tr0njavolta) [inline doc/user/content/sql/functions/jsonb_object_agg.md:79]:
```suggestion
In this example, there are multiple values associated with the `k2` key.

The `FILTER` clause in the statement above returns values that are not `NULL` and orders them by the timestamp column to return the most recent associated value.
```

**Review by tr0njavolta — APPROVED** [review]
Very small simplification, but may be clearer in the FILTER explanation section. Also, small nit, but in the `Use a {} to access the value for a specific key:`, could we use "return" instead of access?

Approving so I don't block you. Thanks for tackling this!

File: doc/user/data/sql_funcs.yml [hunk 2]
```diff
@@ -550,7 +553,7 @@
     url: "/docs/sql/types/jsonb/#jsonb_array_length"
 
   - signature: 'jsonb_build_array(x: ...) -> jsonb'
-    description: The elements of `x` in a `jsonb` array. Elements can be of heterogenous
+    description: Outputs each element of `x` as a `jsonb` array. Elements can be of heterogenous
```
Comment (morsapaes) [inline doc/user/data/sql_funcs.yml:556]:
```suggestion
    description: Output each element of `x` as a `jsonb` array. Elements can be of heterogenous
```

File: doc/user/data/sql_funcs.yml [hunk 3]
```diff
@@ -567,6 +570,9 @@
     description: "`j`'s outermost elements if `j` is an object."
     url: "/docs/sql/types/jsonb/#jsonb_each_text"
 
+  - signature: jsonb_object_agg(keys, values) -> jsonb
+    description: Aggregate keys and values (including nulls) as a jsonb object. ([docs](/sql/functions/jsonb_object_agg))
```
Comment (morsapaes) [inline doc/user/data/sql_funcs.yml:574]:
```suggestion
    description: Aggregate keys and values (including nulls) as a `jsonb` object. ([docs](/sql/functions/jsonb_object_agg))
```

File: doc/user/data/sql_funcs.yml [hunk 4]
```diff
@@ -537,6 +537,9 @@
 
 - type: JSON
   functions:
+  - signature: jsonb_agg(expression) -> jsonb
+    description: Aggregate values (including nulls) as a jsonb array. ([docs](/sql/functions/jsonb_agg))
```
Comment (morsapaes) [inline doc/user/data/sql_funcs.yml:541]:
```suggestion
  - signature: jsonb_agg(expression) -> jsonb
    description: Aggregate values (including nulls) as a jsonb array.
    url:  /docs/sql/functions/jsonb_agg
```

Comment (chuck-alt-delete) [reply doc/user/data/sql_funcs.yml:541]: See comment:

https://github.com/MaterializeInc/materialize/pull/20476#issuecomment-1631893912

Should I still go ahead and use the `url` attribute?

Comment (morsapaes) [reply doc/user/data/sql_funcs.yml:541]: Sounds like we need to fix it, like you suggested, rather than work around it. So — yes, please!

**Pull request merged**